src/theme-provider.js

```diff
@@ -0,0 +1,74 @@
+import React, {PropTypes, Component} from 'react'
+import brcast from 'brcast'
+
+export const CHANNEL = '__glamorous__'
+
+/**
+ * This is a component which will provide a theme to the entire tree
+ * via context and event listener (because pure components block conext updates)
+ * @param {Object} theme the theme object..
+ */
+class ThemeProvider extends Component {
+  broadcast = brcast(this.props.theme)
+
+  // create theme, by merging with outer theme, if present
+  getTheme = passedTheme => {
+    const theme = passedTheme || this.props.theme || {}
+    return {...this.outerTheme, ...theme}
+  }
+
+  getChildContext() {
+    return {
+      [CHANNEL]: this.broadcast,
+    }
+  }
+
+  setOuterTheme = theme => {
+    this.outerTheme = theme
```

Comment:
Yeah, we should probably initialize this above. I'll do that really quick. Wanna get this merged :)